### PR TITLE
Use QgsFileWidget in several dialogs

### PR DESCRIPTION
**Pull request by nirvn**
## Description
@nyalldawson , this has been on your wishlist for a while, quick review appreciated.

Dialogs updated:
- Create shapefile layer
- Create geopackage layer
- Save vector layer as

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [ ] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [ ] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [ ] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

**Comment by nyalldawson** [comment]
Looking good! One thing i'd like to ensure - one problem with the existing widgets which do this is that they often default to no folder - meaning that entering "my_file.shp" in the line edit tries to save to the qgis bin folder, which is 100% what no-one wants.

Can we make sure the default folder is set to something reasonable if not manually set for the widget?

File: src/gui/qgsfilewidget.h [hunk 1]
```diff
@@ -120,6 +121,17 @@ class GUI_EXPORT QgsFileWidget : public QWidget
      */
     void setFilter( const QString &filter );
 
+    /**
+     * Sets whether a confirmation to overwrite an existing file will appear
+     * \param confirmOverwrite If set to true, an overwrite confirmation will be shown
```
Comment (m-kuhn) [inline src/gui/qgsfilewidget.h]: Should the default value of this property be documented?

**Comment by nirvn** [comment]
@nyalldawson , yeah, the nice thing about moving all these to QgsFileWidget is that once it's done, dealing with that will be applied across the board.

IMHO, a filename such as "z.shp" should become relative to the project path. That would match other parts of QGIS and actually be quite cool (for e.g., it'd play very well with the browser panel's project tree item).

I'm keen to merge this and work on the relative path issue in another PR.

File: src/gui/qgsfilewidget.h (hunk 1 above)
Comment (nirvn) [reply src/gui/qgsfilewidget.h]: @m-kuhn , done.

Comment (m-kuhn) [reply src/gui/qgsfilewidget.h]: Thanks.

FYI, Personally I started to document such setter/getter pairs as properties and just copy paste the content. Without an extra description for the setter parameter which actually only states the obvious.

Makes them much easier to write and focus on what the *property* does and not what the *setter* does (except of course the setter has some very special magic in it).

```
Defines whether... Defaults to...
```

**Pull request merged**